src/actions.ml

```diff
@@ -304,167 +306,161 @@ let fetch_bench_results ~job_info () =
-        else Lwt.return_error (f "parsing %s table." table_name)
-      in
-      let* slow_number = parse_quantity slow_table "slow" in
-      let* fast_number = parse_quantity fast_table "fast" in
-      match (slow_number, fast_number) with
-      | Error e, _ | _, Error e ->
-          Lwt.return_error (f "Fetch bench regex issue: %s" e)
-      | Ok slow_number, Ok fast_number ->
-          Lwt.return_ok
-            {summary_table; slow_table; slow_number; fast_table; fast_number} )
+  let get_table_data name = artifact_url name |> fetch_artifact in
+  let parse_quantity table =
+    let regexp = {|.*TOP \([0-9]*\)|} in
+    if Helpers.string_match ~regexp table then
+      Some (Int.of_string @@ Str.matched_group 1 table)
+    else None
+  in
+  let open Lwt_result.Syntax in
+  let get_table name =
+    let* table = get_table_data name in
+    let count = parse_quantity table in
+    Lwt_result.return BenchResults.Table.{table; count}
+  in
+  let* summary_table = get_table "bench_summary" in
+  let* slow_table = get_table "slow_table" in
+  let* fast_table = get_table "fast_table" in
+  let* slow_table_pdiff = get_table "slow_table_pdiff" in
+  let* fast_table_pdiff = get_table "fast_table_pdiff" in
+  Lwt_result.return
+    BenchResults.
+      {summary_table; slow_table; fast_table; slow_table_pdiff; fast_table_pdiff}
 
-let bench_text = function
-  | Ok results ->
-      (* Formatting helpers *)
-      let header2 str = f "## %s" str in
-      let code_wrap str = f "```\n%s\n```" str in
-      (* Document *)
-      let open BenchResults in
-      [ header2 ":checkered_flag: Bench Summary:"
-      ; code_wrap results.summary_table
-      ; header2 @@ f ":turtle: Top %d slow downs:" results.slow_number
-      ; code_wrap results.slow_table
-      ; header2 @@ f ":rabbit2: Top %d speed ups:" results.fast_number
-      ; code_wrap results.fast_table ]
-      |> String.concat ~sep:"\n" |> Lwt.return
-  | Error e ->
-      f "Error occured when creating bench summary: %s\n" e |> Lwt.return
+let bench_text results =
+  (* Formatting helpers *)
+  let header2 str = f "## %s" str in
+  let code_wrap str = f "```\n%s\n```" str in
+  let pp_count () count =
+    match count with None -> "" | Some n -> Printf.sprintf "Top %d" n
+  in
+  let pp_table title table ~emoji =
+    [ header2
+      @@ f "%s %a %s:" emoji pp_count table.BenchResults.Table.count title
+    ; code_wrap table.table ]
+  in
+  (* Document *)
+  let BenchResults.
+        { summary_table
+        ; slow_table
+        ; fast_table
+        ; slow_table_pdiff
+        ; fast_table_pdiff } =
+    results
+  in
+  [ pp_table "Bench Summary" summary_table ~emoji:":checkered_flag:"
+  ; pp_table "Slow Downs" slow_table ~emoji:":turtle:"
+  ; pp_table "Speed Ups" fast_table ~emoji:":rabbit2:"
+  ; pp_table "Slow Downs (%)" slow_table_pdiff ~emoji:":snail:"
+  ; pp_table "Speed Ups (%)" fast_table_pdiff ~emoji:":eagle:" ]
+  |> Caml.List.flatten |> String.concat ~sep:"\n"
```

Comment:
What is "Caml"?

Reply:
It has been a while, but I believe that is Base reexporting the stdlib since it shadows List. I think flatten wasn't available.

Reply:
OK. Then, an alternative way to access it that is already used in a few places is through the explicit `Stdlib` module.

Reply:
I've stopped using it all together now.